poetry/core/version/grammars/parser.py

```diff
@@ -1,26 +1,20 @@
-import os
+from pathlib import Path
 
-from typing import TYPE_CHECKING
-from typing import Optional
+from lark import Lark
 
 
-if TYPE_CHECKING:
-    from lark import Lark  # noqa
-    from lark import Tree  # noqa
+GRAMMAR_DIR = Path(__file__).parent
 
+# Parser: PEP 440
+# we use earley because the grammar is ambiguous
+PARSER_PEP_440 = Lark.open(GRAMMAR_DIR / "pep440.lark", parser="earley", debug=False)
 
-class Parser:
-    def __init__(self, grammar: str) -> None:
-        self._grammar = grammar
-        self._parser: Optional["Lark"] = None
+# Parser: PEP 508 Constraints
+PARSER_PEP_508_CONSTRAINTS = Lark.open(
+    GRAMMAR_DIR / "pep508.lark", parser="lalr", debug=False
+)
 
-    def parse(self, string: str) -> "Tree":
-        from lark import Lark
-
-        if self._parser is None:
-            self._parser = Lark.open(
-                os.path.join(os.path.dirname(__file__), f"{self._grammar}.lark"),
-                parser="lalr",
-            )
-
-        return self._parser.parse(string)
```

Comment:
We need this class to lazily load grammers, otherwise the performance of Poetry will be severely degraded.

Reply:
Can you make the changes to add it back?

Reply:
Also, I don't think this is the right place to instantiate the parsers.

Reply:
Sure. I don't have a strong opinion on this. I was keeping the parser here since the original one was here. 

Happy to move to a factory model. That said, these will not be loaded until these are imported anyway, and they are imported when they are needed only. So in a way they are already lazy loaded and shouldn't have any change from the previous case.